src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java

```diff
@@ -0,0 +1,259 @@
+
+  List<List<String>> table;
+  List<SampleDto> allPreciseIdentities;
+
+  private static final List<ColumnDefinition> COLUMNS =
+      Collections.unmodifiableList(
+          Arrays.asList(
+              new ColumnDefinition("ID"),
+              new ColumnDefinition("Screening/Randomization Serum"),
+              new ColumnDefinition("Screening/Randomization Plasma"),
+              new ColumnDefinition("Screening/Randomization Buffy Coat"),
+              new ColumnDefinition("Screening/Randomization Urine Supernatant"),
+              new ColumnDefinition("Screening/Randomization Urine Pellet"),
+              new ColumnDefinition("6 mo Follow Up Serum"),
+              new ColumnDefinition("6 mo Follow Up Plasma"),
+              new ColumnDefinition("6 mo Follow Up Buffy Coat"),
+              new ColumnDefinition("1 yr Follow Up Serum"),
+              new ColumnDefinition("1 yr Follow Up Plasma"),
+              new ColumnDefinition("1 yr Follow Up Buffy Coat"),
+              new ColumnDefinition("1 yr Follow Up Urine Supernatant"),
+              new ColumnDefinition("1 yr Follow Up Urine Pellet"),
+              new ColumnDefinition("18 mo Follow Up Serum"),
+              new ColumnDefinition("18 mo Follow Up Plasma"),
+              new ColumnDefinition("18 mo Follow Up Buffy Coat"),
+              new ColumnDefinition("2 yr Follow Up Serum"),
+              new ColumnDefinition("2 yr Follow Up Plasma"),
+              new ColumnDefinition("2 yr Follow Up Buffy Coat"),
+              new ColumnDefinition("2 yr Follow Up Urine Supernatant"),
+              new ColumnDefinition("2 yr Follow Up Urine Pellet"),
+              new ColumnDefinition("Bx Slides - Positive"),
+              new ColumnDefinition("Bx Slides - Negative"),
+              new ColumnDefinition("RP Slides")));
+
+  private enum TimePoint {
+    // Coded by the Times Received portion of the tissue name: PRE_1111111_Xx_X_nn_*#*-1
+    RANDOMIZATION("Screening/Randomization", 1),
+    SIX("6 mo Follow Up", 2),
+    TWELVE("1 yr Follow Up", 3), //
+    EIGHTEEN("18 mo Follow Up", 4),
+    TWENTY_FOUR("2 yr Follow Up", 5),
+    BX("Bx Slides", 6),
+    RP("RP Slides", 7);
+    private final String key;
+    private final int timePointCode;
+    private static final Map<String, PreciseCaseReport.TimePoint> lookup = new TreeMap<>();
+
+    static {
+      for (PreciseCaseReport.TimePoint s : EnumSet.allOf(PreciseCaseReport.TimePoint.class)) {
+        lookup.put(s.getKey(), s);
+      }
+    }
+
+    TimePoint(String key, int timePointCode) {
+      this.key = key;
+      this.timePointCode = timePointCode;
+    }
+
+    public String getKey() {
+      return key;
+    }
+
+    /**
+     * Assumes that PRECISE samples will always have _nn_ in the name, and that Tissue Origin will
+     * never be "nn". Should hopefully be a safe assumption since this is a biobanking project where
+     * tracking tissue origin is critical.
```

Comment:
I don't think anything depends on this assumption anymore?

Reply:
Fixed!